File: backend/app/schemas/source.py
from datetime import datetime

from pydantic import BaseModel


class SourceBase(BaseModel):
    name: str
    url: str
    source_type: str
    local_folder: str
    audio_format: str = "mp3"
    name_format: str | None = None
    sync_enabled: bool = True
    original_art: bool = True
    extract_artist: bool = False


class SourceCreate(SourceBase):
    pass


class SourceUpdate(BaseModel):
    name: str | None = None
    url: str | None = None
    source_type: str | None = None
    local_folder: str | None = None
    audio_format: str | None = None
    name_format: str | None = None
    sync_enabled: bool | None = None
    original_art: bool | None = None
    extract_artist: bool | None = None


class SourceRead(SourceBase):
    id: int
    created_at: datetime
    updated_at: datetime
    last_sync_status: str | None = None
    last_sync_at: datetime | None = None

    model_config = {"from_attributes": True}

File: frontend/src/types/source.ts
export interface Source {
  id: number;
  name: string;
  url: string;
  source_type: string;
  local_folder: string;
  audio_format: string;
  name_format: string | null;
  sync_enabled: boolean;
  original_art: boolean;
  extract_artist: boolean;
  created_at: string;
  updated_at: string;
  last_sync_status: string | null;
  last_sync_at: string | null;
}

export interface SourceCreate {
  name: string;
  url: string;
  source_type: string;
  local_folder: string;
  audio_format?: string;
  name_format?: string | null;
  sync_enabled?: boolean;
  original_art?: boolean;
  extract_artist?: boolean;
}

export type SourceUpdate = Partial<SourceCreate>;

export interface TrackFile {
  name: string;
  relative_path: string | null;
  size: number;
  modified_at: string | null;
  status: "synced" | "missing" | "untracked";
  track_id: string | null;
}
